### PR TITLE
Use API v2 for Bark notification

### PR DESCRIPTION
**Pull request by harryzcy**
Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Bark notification provider offers a API v2 and should be preferred ([see its doc](https://github.com/Finb/bark-server/blob/master/docs/API_V2.md)). The new API uses standard POST requests to send notifications.

## Type of change

Please delete any options that are not relevant.

- Other

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

### CODE REVIEW

**Comment by louislam** [comment]
Will it affects current users whose bark server is API v1?

**Comment by harryzcy** [comment]
No. In the official [bark project](https://github.com/Finb/bark-server), its API v2 was added in Feb, 2021, even before Uptime Kuma supports bark notifications (in Oct 2021). So all users using Bark should already have API v2 support. It won't break anything for current users.

**Comment by louislam** [comment]
I am still concerning, because users could possibly be using the old version of bark-server?

If yes, it will be a breaking change and the current implementation is a better handling for both old and new bark-servers.

cc: @Lakr233

**Comment by Lakr233** [comment]
I don't even know there is a v2 API of bark server.................... 🥲

we could put a notice? > Bark API v2 is required?

**Comment by harryzcy** [comment]
If we have to keep API v1 version, then I think it's good to introduce API v2 as a separate provider. 
What do you think @louislam

**Review by Lakr233 — APPROVED** [review]

**Comment by louislam** [comment]
> If we have to keep API v1 version, then I think it's good to introduce API v2 as a separate provider. What do you think @louislam

A separate provider may be too much. You can add a dropdown for selecting V1/V2 api. 


- Notifications that setup before this pr:
Keep using V1 

- New notifications:
Default value = V2

**Comment by harryzcy** [comment]
> > If we have to keep API v1 version, then I think it's good to introduce API v2 as a separate provider. What do you think @louislam
> 
> A separate provider may be too much. You can add a dropdown for selecting V1/V2 api.
> 
> * Notifications that setup before this pr:
>   Keep using V1
> * New notifications:
>   Default value = V2

Sure, it's now implemented

File: src/components/notifications/Bark.vue [hunk 1]
```diff
@@ -1,4 +1,11 @@
 <template>
+    <div class="mb-3">
+        <label for="Bark Endpoint" class="form-label">{{ $t("Bark API Version") }}</label>
+        <select id="Bark Sound" v-model="$parent.notification.apiVersion" class="form-select" required>
```
Comment (chakflying) [inline src/components/notifications/Bark.vue:4]:
```suggestion
        <label for="Bark API Version" class="form-label">{{ $t("Bark API Version") }}</label>
        <select id="Bark API Version" v-model="$parent.notification.apiVersion" class="form-select" required>
```

**Review by chakflying — CHANGES_REQUESTED** [review]
Also, new translated string should be added to `en.json`.

File: server/notification-providers/bark.js [hunk 2]
```diff
@@ -96,12 +95,23 @@ class Bark extends NotificationProvider {
...
      */
     async postNotification(notification, title, subtitle, endpoint) {
-        // url encode title and subtitle
-        title = encodeURIComponent(title);
-        subtitle = encodeURIComponent(subtitle);
-        let postUrl = endpoint + "/" + title + "/" + subtitle;
-        postUrl = this.appendAdditionalParameters(notification, postUrl);
-        let result = await axios.get(postUrl);
+        let result;
+        if (notification.apiVersion == "v1") {
```
Comment (chakflying) [inline server/notification-providers/bark.js:99]:
```suggestion
        if (notification.apiVersion === "v1" || notification.apiVersion == null) {
```

**Review by chakflying — CHANGES_REQUESTED** [review]
If users created the notification in an older version, the value would be null? I think added a check here would make sure that they continue to use v1.

**Pull request merged**